src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -88,48 +86,62 @@ internal void LogDiagnostic(Diagnostic diagnostic, CultureInfo culture)
 
         private void WriteLocations(Location location, IReadOnlyList<Location> additionalLocations)
         {
-            _writer.WriteArrayStart("locations");
+            if (location.SourceTree != null)
+            {
+                _writer.WriteArrayStart("locations");
+                _writer.WriteObjectStart(); // location
+                _writer.WriteKey("analysisTarget");
```

Comment:
I agree with Larry here. 

> For Roslyn, I tend to think that the project file may be the analysisTarget.
Note that there is no project file known at this layer (and the analysis could well have been kicked off without one even existing). All we have is a compilation which effectively captures a csc command line as Larry wrote. There is a 1:1 between compilations and runs and I still don't understand the value of repeating the same analysisTarget for every result in the run.